flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/SavepointObserver.java

```diff
@@ -131,9 +136,56 @@ private Optional<String> observeTriggeredSavepointProgress(
 
         LOG.info("Savepoint status updated with latest completed savepoint info");
         currentSavepointInfo.updateLastSavepoint(savepointFetchResult.getSavepoint());
+        updateSavepointHistory(
+                currentSavepointInfo, savepointFetchResult.getSavepoint(), deployedConfig);
```

Comment:
I think we should also update the savepoint history when we shut down using savepoint. (Or in case of failure of the shutdown operation as part of the `observeLatestSavepoint`)

Reply:
The job needs to be running for savepoint cleanup to succeed.

Reply:
That makes sense but the savepoint itself could still be recorded in the history and cleaned up later after upgrade